### ISSUE TITLE
Ignore/default methods in the resource interface

### ISSUE THREAD

**Issue by codingchili** [eclipse-ee4j/jersey#4183]
Hello,

Is there any way to ignore methods declared in a resource interface? The use case is adding a default method that catches a NotFoundException and returns an Optional.

Would prefer to not have to catch JAX-RS specific exceptions in the business layer and also avoid checked exceptions by using an Optional here, example:

```
@Path("/")
interface RestService {
    @Path("/string") @GET
    String getString();

    default Optional<String> getStringOptional() {
        try {
            return Optional.of(getString());
        } catch (NotFoundException e) {
            return Optional.empty();
        }
    }
}
```
I don't see why default methods are being proxied. The example fails because getStringOptional is not a "resource method", if resource annotations are added then a proxy is created and the default method never called.

The initial approach was to create a ClientResponseFilter that threw a custom 404 exception that isn't tied to the JAX-RS spec. This didn't work because we're using the CXF client and any exceptions thrown from a filter is wrapped. 

Have not tried throwing exceptions from a filter when using the Jersey client, I'd like to avoid that because I got even more issues with it, compared to CXF. JNDI lookups failing etc.. not the rabbit hole I wanted to explore.

**Comment by jansupol** [comment]
What do you use to create the `RestService` proxy? Is it `jersey-proxy-client` or `jersey-mp-rest-client`?

**Comment by codingchili** [comment]
`jersey-proxy-client`

```
Client resource = JerseyClientBuilder.newClient(new ClientConfig());
return WebResourceFactory.newResource(apiClass, resource.target("url.."));
```

**Comment by ivangreene** [comment]
Agree, WebResourceFactory should pass through to the default method body. How should it respond to a default interface method with an http method annotation though? I can't think of a reason for a default method with an http method annotation but should probably determine the behavior before implementing a fix

**Comment by codingchili** [comment]
Maybe you want to do a custom call (not using the proxy client) from that method and want to use annotations to make that call somehow. A long shot.

What would someone expect happen? I would expect all annotations to be completely ignored. Add an api annotation and the default method will not be called anymore, doesn't seem intuitive. Either that or throw an exception, but that's going to be runtime - and I think it's an unnecessary restriction.

**Comment by jansupol** [comment]
The use case I see for the default method being proxied is that the default method can be a resource method or a sub-resource locator.
